tests/test_filelock.py

```diff
@@ -0,0 +1,29 @@
+import time
+import os
+
+import pytest
+
+from datasets.utils.filelock import FileLock, Timeout
+
+
+def test_filelock(tmpdir):
+    lock1 = FileLock(tmpdir / "foo.lock")
+    lock2 = FileLock(tmpdir / "foo.lock")
+    timeout = 0.01
+    with lock1.acquire():
+        with pytest.raises(Timeout):
+            _start = time.time()
+            lock2.acquire(timeout)
+            assert time.time() - _start > timeout
+
+
+def test_long_filename(tmpdir):
+    filename = "a" * 1000 + ".lock"
+    lock1 = FileLock(tmpdir / filename)
+    assert lock1._lock_file.endswith(".lock")
+    assert not lock1._lock_file.endswith(filename)
+    assert len(os.path.basename(lock1._lock_file)) <= 255
+    lock2 = FileLock(tmpdir / filename)
+    with lock1.acquire():
+        with pytest.raises(Timeout):
```

Comment:
Increased latency by 0.01 two times?

We are already facing too-long test suite issue...

Reply:
This is nothing compared to the tests that take several seconds